### PR TITLE
[ROCm] Update ROCm runner labels to new labels

### PR DESCRIPTION
**Pull request by psanal35**
This PR updates ROCm CI runner labels from `linux-x86-64-Ngpu-amd` to `amd-do-linux.jax.gpu.gfx950.N`.

### CODE REVIEW

File: ci/upload_rocm_logs.sh [hunk 1]
```diff
@@ -47,9 +47,9 @@ RUN_STARTED_AT="$(
 DATE="${RUN_STARTED_AT%%T*}"
 [[ -n "${DATE}" ]] || DATE="$(date -u +%F)"
 
-# GPU count from runner name (e.g. linux-x86-64-8gpu-amd -> 8).
+# GPU count from runner name (e.g. amd-do-linux.jax.gpu.gfx950.8 -> 8).
 GPU_COUNT=""
-if [[ "${INPUT_RUNNER}" =~ ([0-9]+)gpu ]]; then
+if [[ "${INPUT_RUNNER}" =~ gfx950\.([0-9]+) ]]; then
```
[the remark this answers is not in the document]
Comment (psanal35) [reply ci/upload_rocm_logs.sh]: done~

**Review by magaonka-amd — APPROVED** [review]
thanks , LGTM

**Review by hawkinsp — APPROVED** [review]
